### ISSUE TITLE
Improve keyboard support

### ISSUE THREAD

**Issue by troxler** [olton/Metro-UI-CSS#1366]
Some form components have no or poor keyboard support. They cannot be focussed by using the keyboard and/or their values cannot be changed with the keyboard only.

Those are some components which have a lacking keyboard support:

- checkbox
- radio
- select
- spinner
- switch
- calendar
- calendar picker
- date picker
- time picker
- dialogs
- info box

In addition, the focus styling of certain components should be improved as it is barely visible (for example in the slider component or `<button class="button link">`).

Reference: https://github.com/troxler/awesome-css-frameworks/pull/30

**Comment by olton** [comment]
If you need additional component functionality, you can create one issue for each component with a detailed description of component behavior, user interactions or style changes.

**Comment by troxler** [comment]
I'm not involved in this project and I don't plan to do so. So I won't create individual issues as my only intention was to mention that currently this framework is a little lacking in that area.

There are already (pseudo) standards and proven patterns for many different widgets. You can find some here: https://www.w3.org/TR/wai-aria-practices/#aria_ex

For example, the [checkbox](https://www.w3.org/TR/wai-aria-practices/#checkbox):

> When the checkbox has focus, pressing the Space key changes the state of the checkbox.

**Comment by olton** [comment]
In Metro 4 chackbox, radio, switch can change state when user press Space

**Comment by troxler** [comment]
Oh, yes you are right. But I do not see which widget is currently focussed which makes it hard to actually use the keyboard.

**Comment by olton** [comment]
Thank you, I will review all input elements and add an attribute that, when activated (example: data-focus="true"), will display the outline with focus.

**Comment by petrk94** [comment]
Is the issue finally solved?
